torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -1203,11 +1203,21 @@ def test_3_level_hierarchical_model_averager(self):
             averager = hierarchicalSGD.HierarchicalModelAverager(
                 period_group_size_dict=period_group_size_dict, warmup_steps=warmup_steps
             )
+            self.assertEqual(dist.get_pg_count(), len(period_group_size_dict))
+
             subgroup1 = averager.period_process_group_dict[subgroup_avg_period1]
             subgroup2 = averager.period_process_group_dict[subgroup_avg_period2]
+            subgroup1_name = dist._get_process_group_name(subgroup1)
+            subgroup2_name = dist._get_process_group_name(subgroup2)
+            real_group_ranks_res1 = []
+            real_group_ranks_res2 = []
+            pg_configs = dist.get_pg_config()
+            for pg_config in pg_configs:
+                if pg_configs['pg_name'] == subgroup1_name:
+                    real_group_ranks_res1 = pg_configs['ranks']
+                elif pg_configs['pg_name'] == subgroup2_name:
+                    real_group_ranks_res2 = pg_configs['ranks']
```

Comment:
This for loop and if check seems to suggest that List-form return may not be the most ideal form.